Link to more existing examples from new examples page (#637)
dfinity/docs: modules/examples/pages/index.adoc

In addition to our tutorials section, here are some quick references! You can also check out our https://github.com/dfinity/examples[Examples Repo] or https://github.com/dfinity/awesome-dfinity[Awesome Dfinity], a curated set of open-source projects and examples from our community!

[[motoko]]

> == Motoko

Author: ```suggestion
== Motoko Examples
```

[[motoko-playground]]
=== Motoko Playground

Some (but not all) of the following Motoko examples are available via the https://m7sm4-2iaaa-aaaab-qabra-cai.raw.ic0.app/[Motoko Playground]. The playground provides the simplest enviromnent for trying out pure Motoko examples without having to download and learn to use the {SDK}, but does not support applications with frontends.

=== Basic

- https://github.com/dfinity/examples/tree/master/motoko/actor_reference[actor_reference] -- IC management canister as an actor (reference).
- https://github.com/dfinity/examples/tree/master/motoko/counter[counter] -- basic (stable) variable demo.
- https://github.com/dfinity/examples/tree/master/motoko/echo[echo] -- basic query function demo.
- https://github.com/dfinity/examples/tree/master/motoko/factorial[factorial] -- basic factorial demo.
- https://github.com/dfinity/examples/tree/master/motoko/hello-world[hello-world] -- basic hello world demo.

> - https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[hello_cycles] -- basic cycle transfer demo.

Reviewer: ```suggestion
- https://github.com/dfinity/examples/tree/master/motoko/hello_cycles[hello_cycles] -- Receive and transfer cycles and check your balance.
```

> - https://github.com/dfinity/examples/tree/master/motoko/whoami[whoami] -- basic caller identification demo.

Reviewer: ```suggestion
- https://github.com/dfinity/examples/tree/master/motoko/whoami[whoami] -- a simple canister that reports the Principal of its caller.
```

=== Intermediate

- https://github.com/dfinity/examples/tree/master/motoko/classes[classes] -- dynamic actor (class) instantiation.
- https://github.com/dfinity/examples/tree/master/motoko/pub-sub[pub-sub]-- multiple canisters, with publisher-subscriber inter-canister calls.
- https://github.com/dfinity/examples/tree/master/motoko/quicksort[quicksort] -- sorting an array, via Quick Sort, in Motoko.
- https://github.com/dfinity/examples/tree/master/motoko/simple-to-do[simple-to-do] -- CRUD-like demo service, sans a front end; see also: phone-book and superheroes.
- https://github.com/dfinity/examples/tree/master/motoko/calc[calc] -- more advanced version of counter demo.

=== Minimal front end.

> These examples use a minimal front end component.

Reviewer: ```suggestion
These examples use a minimal front end component. 
```

- https://github.com/dfinity/examples/tree/master/motoko/random_maze[random_maze] -- random maze generation, with IC-based randomness.
- https://github.com/dfinity/examples/tree/master/motoko/cert-var[cert_var] -- simple certified variable (a single 32-bit number), with client-side certificate validation.

=== Full stack.

These examples use a "conventional" front end component (via `React.Component`).

> - https://github.com/dfinity/examples/tree/master/motoko/life[life] -- demonstrates upgrades among three versions and stable state migration.

Reviewer: ```suggestion
- https://github.com/dfinity/examples/tree/master/motoko/life[life] -- Conway's https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life[Game of Life], running in a Motoko Canister. Demonstrates upgrades among three versions and stable state migration. 
```

- https://github.com/dfinity/examples/tree/master/motoko/calc[calc] -- more advanced version of counter demo.

=== Minimal front end.

These examples use a minimal front end component.

- https://github.com/dfinity/examples/tree/master/motoko/random_maze[random_maze] -- random maze generation, with IC-based randomness.
- https://github.com/dfinity/examples/tree/master/motoko/cert-var[cert_var] -- simple certified variable (a single 32-bit number), with client-side certificate validation.

=== Full stack.

These examples use a "conventional" front end component (via `React.Component`).

- https://github.com/dfinity/examples/tree/master/motoko/life[life] -- Conway's https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life[Game of Life], running in a Motoko Canister. Demonstrates upgrades among three versions and state migration using stable variables. 
- https://github.com/dfinity/examples/tree/master/motoko/phone-book[phone-book] -- CRUD-like demo service.
- https://github.com/dfinity/examples/tree/master/motoko/superheroes[superheroes] -- a simple CRUD dapp. 

> - https://github.com/krpeacock/auth-client-demo[Auth Client Demo] - Simple authentication with Internet Identity

Author: ```suggestion
- https://github.com/krpeacock/auth-client-demo[Auth Client Demo] - simple authentication with Internet Identity.
```

> - https://github.com/krpeacock/ic-avatar[IC-Avatar] - Full-stack Motoko + React with more advanced authentication, image uploads, and optimistic updates

Author: ```suggestion
- https://github.com/krpeacock/ic-avatar[IC-Avatar] - full-stack Motoko + React with more advanced authentication, image uploads, and optimistic updates.
```

[[motoko-advanced]]
=== Advanced Examples

Large examples contributed by the community and not necessarily up-to-date:

> - Reversi game, in Motoko https://github.com/ninegua/reversi[src] accompanying https://ninegua.github.io/reversi[Blog post]

Author: ```suggestion
- Reversi game, in Motoko https://github.com/ninegua/reversi[src] accompanying https://ninegua.github.io/reversi[Blog post].
```

- QR code generator in Motoko https://github.com/enzoh/motoko-qr[src] accompanying

>   https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article]

Author: ```suggestion
  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article].
```

>   https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865bB[Medium article]

Author: https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b
```suggestion
  https://medium.com/@ehaussecker/my-first-microservice-on-dfinity-3ac5c142865b[Medium article]
```

> - https://github.com/timohanke/icvault[IC Vault] - End-to-end Encryption

Author: ```suggestion
- https://github.com/timohanke/icvault[IC Vault] - end-to-end Encryption.
```

- https://www.youtube.com/watch?v=36L33S_DYHY&ab_channel=DFINITY[Community Conversations | Best Practices for Canisters in Rust]

More Rust examples coming soon!

> 

Reviewer 2: There is also a counter example in wat.

Author: Ok, I'll reference that.

== C

For examples of projects written in the C programming language for the {IC}, see link:https://github.com/dfinity/examples/tree/master/c[C++ sample projects].

WARNING: these examples are likely to bit-rot and may need updating and are provided for illustration only.

* link:https://github.com/dfinity/examples/tree/master/c/adventure[Adventure game]
* link:https://github.com/dfinity/examples/tree/master/c/qr[QR code generator]
* link:https://github.com/dfinity/examples/tree/master/c/reverse[Reverse game]

> 

Author: ```suggestion

== Wat (Wasm textual format)

Just for fun, we also have a simple example written in pure Wasm textual format:

* link:https://github.com/dfinity/examples/tree/master/wasm/counter[counter]

```
